app/controllers/api_controller.rb

```diff
@@ -35,6 +36,24 @@ def user_for_paper_trail
 
 private
 
+  def idempotent_action
```

Comment:
`ApiController` is already huge and filled with things - perhaps you would consider a composition pattern for this instead and move these methods plus the `before_action` and `around_action` calls in the relevant controller to a new `Idempotable` concern? That way all the idempotency related code is in one place.

Reply:
Yep, that is a fair point, I'll have a go at moving this into a concern 👍

Reply:
now moved to a concern